src/storage.rs

```diff
@@ -61,26 +62,58 @@ where
                 .for_each(|name| trace!("{}", name.file_name().as_os_str().to_str().unwrap())); // @TODO handle unwrap explicitly
             self.init_from_existing(files_in_work_dir).unwrap(); // @TODO handle unwrap explicitly
         }
-        // @TODO implement
+        self.initialized = Some(());
         Ok(())
     }
 
-    /// Description
-    /// Examples
+    /// # Description
+    /// Writes bytes `value` to active blob
+    /// If active blob reaches it limit, create new and close old
+    /// Returns number of bytes, written to blob
+    /// # Examples
 
     // @TODO specify more useful error type
-    pub fn write(&mut self) -> Result<(), ()> {
-        // @TODO implement
-        Ok(())
+    pub fn write(&mut self, key: K, value: &[u8]) -> Result<usize, ()> {
+        self.initialized.ok_or(())?;
+        // @TODO process unwrap explicitly
+        if self.active_blob.as_ref().unwrap().size()? + value.len()
+            > self.config.max_blob_size.unwrap()
+            || self.active_blob.as_ref().unwrap().len()? >= self.config.max_data_in_blob.unwrap()
```

Comment:
`active_blob` should be read inside critical section to prevent torn read. If you make it a pointer then critical section can be avoided.